core/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -123,7 +123,9 @@ public synchronized ClientStream newStream(
       return new NoopClientStream() {
         @Override
         public void start(ClientStreamListener listener) {
-          listener.closed(shutdownStatus, new Metadata());
+          synchronized (InProcessTransport.this) {
+            listener.closed(shutdownStatus, new Metadata());
```

Comment:
Ah, I see. Go ahead and save the shutdownStatus within newStream as a final local variable, then.